benchmarks/benchmarks/bench_function_base.py

```diff
@@ -95,6 +95,17 @@ def time_select_larger(self):
         np.select(self.cond_large, ([self.d, self.e] * 10))
 
 
+def memoize(f):
+    _memoized = {}
+    def wrapped(*args):
+        if args not in _memoized:
+            _memoized[args] = f(*args)
+        
+        return _memoized[args]
```

Comment:
I think you need a `.copy()` here

Reply:
At least, you do if you want to time the in-place sort instead of the out-of-place one

Reply:
Done, thanks. I tested even with the `.copy()`, but it seems that it still gives flat results, so I resorted to using `np.sort(arr, ...)` instead of `arr.sort(...)`. I suspect it's only preparing the data once per test, but using it multiple times in the same test without a deep copy/calling `setup` all over again.

Reply:
> I suspect it's only preparing the data once per test, but using it multiple times in the same test

Sounds like the right diagnosis. Might be worth adding a comment to that effect

Reply:
Added!